File: DSA/NarindVerma_500121898/Day20Q2.py
def merge_arrays(a, b):
    result = []
    i = j = 0
    n1, n2 = len(a), len(b)

    while i < n1 and j < n2:
        if a[i] < b[j]:
            result.append(a[i])
            i += 1
        else:
            result.append(b[j])
            j += 1

    while i < n1:
        result.append(a[i])
        i += 1

    while j < n2:
        result.append(b[j])
        j += 1

    return result

# Example input arrays
a = [1, 3, 5]
b = [2, 4, 6]

# Merging the arrays
merged = merge_arrays(a, b)

# Printing the result
for num in merged:
    print(num, end=' ')

File: DSA/AgamKappal_590011213/day20_q2.c

#include <stdio.h>

void mergeArrays(int a[], int n1, int b[], int n2, int result[]) {
    int i = 0, j = 0, k = 0;

    while (i < n1 && j < n2) {
        if (a[i] < b[j])
            result[k++] = a[i++];
        else
            result[k++] = b[j++];
    }

    while (i < n1)
        result[k++] = a[i++];
    
    while (j < n2)
        result[k++] = b[j++];
}

int main() {
    int a[] = {1, 3, 5};
    int b[] = {2, 4, 6};
    int n1 = sizeof(a) / sizeof(a[0]);
    int n2 = sizeof(b) / sizeof(b[0]);
    int result[n1 + n2];

    mergeArrays(a, n1, b, n2, result);

    for (int i = 0; i < n1 + n2; i++) {
        printf("%d ", result[i]);
    }

    return 0;
}
